client/examples/perf.rs

```diff
@@ -0,0 +1,466 @@
+
+            loop {
+                let now = Instant::now();
+                if now > *start + duration {
+                    let overshoot = now - (*start + duration);
+                    *start = now - overshoot;
+
+                    break;
+                }
+
+                let diff = (*start + duration - now) / 3;
+                tokio::time::sleep(diff).await;
+            }
+        }
+    }
+
+    /// Execute the plan on one connection
+    async fn test1(
+        request_delay: Option<f32>,
+        client: &Client,
+        plan: &TaskPlan,
+    ) -> anyhow::Result<Duration> {
+        let start = Instant::now();
+
+        let mut start_limiter = Instant::now();
+        for Task { request, count } in &plan.tasks {
+            for _ in 0..*count {
+                limit_rate(&mut start_limiter, request_delay).await;
+                match request {
+                    Request::FindNode => {
+                        client.find_node(client.node_id()).await?;
+                    }
+                    Request::Neighbours(n) => {
+                        client.neighbours(*n).await?;
+                    }
+                    Request::Ping => {
+                        client.ping_sessions().await;
+                    }
+                }
+            }
+        }
+
+        let elapsed = Instant::now() - start;
+
+        Ok(elapsed)
+    }
+
+    /// Plan tasks according to the user specification
+    fn plan_tasks(load_cmd: &LoadCommand) -> TaskPlan {
+        let mut tasks = Vec::new();
+
+        let find_node_weight = load_cmd.find_node_weight;
+        let neighbours_weight = load_cmd.neighbours_weight;
+        let ping_weight = load_cmd.ping_weight;
+
+        let all_count = find_node_weight + neighbours_weight + ping_weight;
+        if all_count == 0 {
+            log::error!("At least one request type must have non-zero weight!");
+            panic!();
+        }
+
+        let scale = load_cmd.requests_per_connection.get() / all_count;
+
+        let find_count = find_node_weight * scale;
+        let neighbour_count = neighbours_weight * scale;
+        let ping_count = ping_weight * scale;
+
+        tasks.push(Task {
+            request: Request::FindNode,
+            count: find_count,
+        });
+        tasks.push(Task {
+            request: Request::Neighbours(load_cmd.neighbours_size),
+            count: neighbour_count,
+        });
+        tasks.push(Task {
+            request: Request::Ping,
+            count: ping_count,
+        });
+
+        TaskPlan { tasks }
+    }
+
+    /// Load test core
+    async fn measure_load(args: &Options, load_cmd: &LoadCommand) -> anyhow::Result<Measurements> {
+        let mut measurements = Measurements::new();
+        let mut clients = Vec::new();
+        for target_connections in &load_cmd.connections {
+            // Drop or create connections to reach the target
+            if *target_connections > clients.len() {
```

Comment:
Clippy suggests rewriting these ifs to match i.e. 
```rust

            match target_connections.cmp(&clients.len()) {
                std::cmp::Ordering::Less => todo!(),
                std::cmp::Ordering::Equal => todo!(),
                std::cmp::Ordering::Greater => todo!(),
            }
```

Reply:
The cleanest solution turned out to be dropping the `else if ...` and just using an `else`, which also fixes this clippy lint.